src/app.js

```diff
@@ -136,46 +105,68 @@ export const weather = (appId, store, wuser, wpassword, token) =>
           }
 
           // Cancel the action
-          if((astate.action === 'Get_Weather_Conditions' ||
-            astate.action === 'Get_Weather_Forecast') &&
-            next === 'Cancel') {
+          if((astate.intent === 'weather' ||
+            astate.intent === 'forecast') &&
+            intent === 'negation') {
             send(noProblem(user));
 
             // Forget the weather action and city as that was not what the
             // user wanted
-            delete astate.action;
+            delete astate.intent;
             delete astate.city;
             cb(null, astate);
           }
-        });
-      });
-
-    // Handle mentions of entities in messages
-    events.onEntities(req.body, appId, token,
-      (entities, nlp, message, user) => {
 
-        // Run with any previously saved action state
-        state.run(spaceId, user.id, store, (astate, cb) => {
+          // Look for a city in the request, default to last city used
+          const city =
+            cityAndState(focus.extractedInfo.entities) || astate.city;
 
-          // Look for a city and state in the extracted entities
-          const city = cityAndState(entities);
           if(city) {
+            // Remember the city
             astate.city = city;
-            if(message.id !== astate.message.id)
 
-              // Ask for a confirmation to get the weather conditions or
-              // weather forecast in the recognized city
-              if(astate.action === 'Get_Weather_Conditions')
-                send(confirmConditions(city, user));
+            // Ask the user to confirm
+            if(intent === 'weather')
+              send(confirmConditions(city, user));
 
-              else if(astate.action === 'Get_Weather_Forecast')
-                send(confirmForecast(city, user));
+            else if(intent == 'forecast')
```

Comment:
I'm surprised that Lint didn't pick that up, but I'd suggest to change == to ===